# HG changeset patch
# User Pantxo Diribarne <pantxo.diribarne@gmail.com>
# Date 1528286824 -7200
#      Wed Jun 06 14:07:04 2018 +0200
# Node ID d8aa0e66aa5c49047bbc739a69732a8323712062
# Parent  6782d1f0fe540dea04f0260c13e65c55b35d3649
open.m: check the requested file exists (bug #54064)

open.m: error out if the file doesn't exist.

diff --git a/scripts/miscellaneous/open.m b/scripts/miscellaneous/open.m
--- a/scripts/miscellaneous/open.m
+++ b/scripts/miscellaneous/open.m
@@ -56,16 +56,18 @@
 ##
 ## Other file types are opened in the appropriate external application.
 ## @end deftypefn
 
 function output = open (file)
 
   if (nargin != 1)
     print_usage ();
+  elseif (! exist (file, "file"))
+    error ("open: unable to find file %s", file);
   endif
 
   if (! ischar (file))
     error ("open: FILE must be a string");
   endif
 
   [~, fname, ext] = fileparts (file);
 
